test/functional/abc_mining_basic.py

```diff
@@ -49,88 +48,42 @@ def assert_getblocktemplate(expected):
             for key, value in expected.items():
                 assert_equal(blockTemplate[key], value)
 
-        # Move block time to just before axion activation
-        node.setmocktime(AXION_ACTIVATION_TIME)
-        node.generatetoaddress(5, address)
-
-        # Before axion activation, the miner fund list is empty
-        assert_getblocktemplate({
-            'coinbasetxn': {
-                'minerfund': {
-                    'addresses': [],
-                    'minimumvalue': 0,
-                },
-            },
-        })
-
-        # Move MTP forward to axion activation
-        node.generatetoaddress(1, address)
-        assert_equal(
-            node.getblockchaininfo()['mediantime'],
-            AXION_ACTIVATION_TIME)
-
         def get_best_coinbase():
             return node.getblock(node.getbestblockhash(), 2)['tx'][0]
 
-        coinbase = get_best_coinbase()
-        assert_equal(len(coinbase['vout']), 1)
-        block_reward = coinbase['vout'][0]['value']
+        block_reward = 50
 
-        # We don't need to test all fields in getblocktemplate since many of
-        # them are covered in mining_basic.py
-        assert_equal(node.getmempoolinfo()['size'], 0)
-        assert_getblocktemplate({
-            'coinbasetxn': {
-                # We expect to start seeing the miner fund addresses since the
-                # next block will start enforcing them.
-                'minerfund': {
-                    'addresses': [MINER_FUND_ADDR],
-                    'minimumvalue': block_reward * 8 // 100 * COIN,
-                },
-            },
-            # Although the coinbase value need not necessarily be the same as
-            # the last block due to halvings and fees, we know this to be true
-            # since we are not crossing a halving boundary and there are no
-            # transactions in the mempool.
-            'coinbasevalue': block_reward * COIN,
-            'mintime': AXION_ACTIVATION_TIME + 1,
-        })
+        # TODO: Cannot use getblocktemplate for block after genesis
```

Comment:
Is this possibly caused by this line? 

```
    if (m_chain.Tip()->nChainWork < nMinimumChainWork) {
        return true;
    }
```

I *really* dislike `::IsInitialBlockDownload()` it caused me some other problems in the past. From my perspective this function is a heuristic mess, and there should be some clear chainstate management at someday.... But not today.

Reply:
Add it to the TODO list, because this actually needs to be fixed before we get on mainnet.